source/fundamentals/typescript.txt

```diff
@@ -67,93 +71,50 @@ You can pass a type parameter that extends the ``Document`` interface like this:
-
-The driver cannot infer the type of values with keys containing **dot
-notation**. Dot notation is a property access syntax for navigating BSON objects.
-Click on the tabs to see code snippets that highlight this behavior:
-
-.. tabs::
-
-  .. tab:: Dot Notation
-     :tabid: dot-notation
-
-     The following code snippet does not raise a type error:
-
-     .. literalinclude:: /code-snippets/typescript/dot-notation.ts
-        :language: typescript
-        :linenos:
-        :start-after: start-no-error
-        :end-before: end-no-error
-
-  .. tab:: Nested Objects
-     :tabid: nested-objects
-
-     The following code snippet raises a type error:
-
-     .. literalinclude:: /code-snippets/typescript/dot-notation.ts
-        :language: typescript
-        :linenos:
-        :start-after: start-error
-        :end-before: end-error
+.. _node-driver-limitations:
 
-     This is the error:
+Limitations For Driver Version {+version+}
+----------------------------------
 
-     .. code-block:: text
+You cannot specify a recursive type as a type parameter in version {+version+} of the driver.
 
-        Type 'string' is not assignable to type 'number'.
+If you specify a recursive type, the driver raises the following error:
 
-Despite the lack of type safety, we still recommend that you use dot notation to
-access nested fields in query and update documents when you use TypeScript. You
-must manually check that your nested field values have your intended type.
+.. code-block:: text
 
-.. note:: Reason To Use Dot Notation
+   error TS2615: Type of property 'r' circularly references itself in mapped type '{ [Key in keyof RecursiveType]: [Key, ...NestedPaths<RecursiveType[Key]>]; }'.
 
-   In the MongoDB Query Language, you must match a subdocument exactly
-   when specifying subdocuments in a query. Dot notation allows you to query
-   nested fields without matching subdocuments exactly.
+A recursive type is a type that references itself. You can update
+the :ref:`Pet <mongodb-node-typescript-pet-interface>` interface
+to be recursive by adding the following field:
 
-   To show this behavior, lets say you have a collection containing
-   only the following document:
-
-   .. code-block:: json
-
-      { field: { s1: "hi", s2: "bye" } }
-
-   The following query returns no results from this collection, as the value of
-   ``field`` does not exactly match ``{ s1: "hi" }``:
-
-   .. literalinclude:: /code-snippets/typescript/note-on-dot-notation.ts
-      :language: typescript
-      :linenos:
-      :start-after: start-no-doc
-      :end-before: end-no-doc
-
-   The following queries both return your document:
+.. code-block:: typescript
+   :emphasize-lines: 2
 
-   .. literalinclude:: /code-snippets/typescript/note-on-dot-notation.ts
-      :language: typescript
-      :linenos:
-      :start-after: start-doc
-      :end-before: end-doc
+   interface Pet {
+      pet: Pet | null;
+      name: string;
+      age: number;
+      cute: true;
+   }
```

Comment:
Do pets have pets? What purpose do the other fields serve, or can they be removed?

Naming aside, is this a more clear way to represent recursion for your example:
```
interface Pet {
  name: string;
  pet?: Pet[];
}
```
https://joshtronic.com/2020/04/19/recursive-types-and-interfaces-in-typescript/

Will leave this up to you (and external review if you seek one).

Reply:
Updated to explain that now a Pet can have a Pet.

The other fields are to match the earlier interface definition. I've removed the "cute" field from both in order to simplify this. I think having 2 fields in the earlier type definition makes sense, as it seems odd to specify an object type with 1 field (an object type must be specified to extend document).

I agree replacing `pet` with `pet?` increases clarity. I do not feel that making `pet` an array increases clarity (seems to just change the meaning of the class from a pet can have one pet to a pet can have many pets).